js/fairdesk.js

```diff
@@ -0,0 +1,1450 @@
+        }
+        if (this.safeString (order, 'status') === 'EXPIRED') {
+            value = 'expired';
+        }
+        if (this.safeString (order, 'status') === 'NEW') {
+            value = 'open';
+        }
+        return value;
+    }
+
+    parseOrder (order, market = undefined) {
+        const symbol = this.safeString (order, 'symbol');
+        const status = this.parseStatus (order); // 'open', 'closed', 'canceled', 'expired', 'rejected'
+        const side = this.safeStringLower (order, 'side');
+        const positionSide = this.safeStringLower (order, 'positionSide');
+        const type = this.parseOrderType (this.safeString (order, 'type'));
+        const price = this.parseNumber (this.safeString (order, 'price') === '--' ? '' : this.safeString (order, 'price'));
+        const amount = this.safeNumber (order, 'origQty');
+        const remaining = amount - this.safeNumber (order, 'executedQty');
+        const timestamp = this.safeInteger (order, 'transactTime');
+        const timeInForce = this.parseTimeInForce (this.safeString (order, 'timeInForce'));
+        return {
+            'info': order,
+            'id': this.safeString (order, 'orderId'),
+            'clientOrderId': this.safeString (order, 'clientOrderId'),
+            'datetime': this.iso8601 (timestamp),
+            'timestamp': timestamp,
+            'lastTradeTimestamp': timestamp,
+            'symbol': symbol,
+            'type': type,
+            'timeInForce': timeInForce,
+            'postOnly': (timeInForce === 'PO'),
+            'side': side,
+            'positionSide': positionSide,
+            'price': price,
+            'stopPrice': this.safeNumber (order, 'triggerPrice'),
+            'amount': amount,
+            'filled': this.safeNumber (order, 'executedQty'),
+            'remaining': remaining,
+            'cost': undefined,
+            'average': undefined,
+            'status': status,
+            'fee': undefined,
+            'trades': undefined,
+        };
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        if (type.toLocaleUpperCase () !== 'MARKET' && type.toLocaleUpperCase () !== 'LIMIT') {
+            throw new ArgumentsRequired (this.id + ' createOrder() only support Market or Limit type');
+        }
+        await this.loadMarkets ();
+        const symbolId = this.parseSymbol (symbol);
+        const market = this.market (symbolId);
+        const sideValue = side.toLocaleUpperCase ();
+        const request = {
+            'symbol': market['id'],
+            'price': price,
+            'quantity': amount,
+            'side': sideValue, // SELL, BUY
+            'type': type.toLocaleUpperCase (), // MARKET, LIMIT
+            'clientOrderId': 'CCXT_' + this.uuid (),
+            'orderRespType': 'ACK',
+            'positionSide': params.positionSide || (sideValue === 'BUY' ? 'LONG' : 'SHORT'),
+            'isolated': params.isolated || false,
```

Comment:
We can't use a dot-notation for the following reason: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#accessing-dictionary-keys